packages/gatsby/src/commands/build-javascript.ts

```diff
@@ -23,7 +26,44 @@ export const buildProductionBundle = async (
   )
 
   return new Promise((resolve, reject) => {
-    webpack(compilerConfig).run((err, stats) => {
+    const compiler = webpack(compilerConfig)
+
+    compiler.hooks.compilation.tap(`webpack-dep-tree-plugin`, compilation => {
+      compilation.hooks.seal.tap(`webpack-dep-tree-plugin`, () => {
+        const state = store.getState()
+        const mapOfTemplatesToStaticQueryHashes = mapTemplatesToStaticQueryHashes(
+          state,
+          compilation
+        )
+
+        mapOfTemplatesToStaticQueryHashes.forEach(
+          (staticQueryHashes, componentPath) => {
+            if (
+              !isEqual(
+                state.staticQueriesByTemplate.get(componentPath)?.sort(),
```

Comment:
You're sorting the items in place here, so the entries in `staticQueriesByTemplate` are mutated. Is this going to be an issue?

Reply:
Excellent point. It shouldn't affect it but I don't like unnecessary mutations either so I'll clean up

Reply:
Fixed